frugen.c

```diff
@@ -674,6 +761,55 @@ int main(int argc, char *argv[])
 					fatal("JSON support was disabled at compile time");
 #endif
 				}
+				else if (use_binary) {
+					int fd = open(optarg, O_RDONLY);
+					if (fd < 0) {
+						fatal("Failed to open file: %s", strerror(errno));
+					}
+
+					struct stat statbuf = {0};
+					if (fstat(fd, &statbuf)) {
+						fatal("Failed to get file properties: %s", strerror(errno));
+					}
+					if (statbuf.st_size > MAX_FILE_SIZE) {
+						fatal("File too large");
+					}
+
+					uint8_t *buffer = calloc(1, statbuf.st_size);
+					if (buffer == NULL) {
+						fatal("Cannot allocate buffer");
+					}
+
+					if (read(fd, buffer, statbuf.st_size) != statbuf.st_size) {
+						fatal("Cannot read file");
+					}
+					close(fd);
+
+					fru_chassis_area_t *chassis_area =
+						find_fru_chassis_area(buffer, statbuf.st_size);
+					if (chassis_area) {
+						if (!fru_decode_chassis_info(chassis_area, &chassis))
+							fatal("Failed to decode chassis");
+						has_chassis = true;
+					}
+					fru_board_area_t *board_area =
+						find_fru_board_area(buffer, statbuf.st_size);
+					if (board_area) {
+						if (!fru_decode_board_info(board_area, &board))
+							fatal("Failed to decode board");
+						has_board = true;
+					}
+
+					fru_product_area_t *product_area =
+						find_fru_product_area(buffer, statbuf.st_size);
+					if (product_area) {
+						if (!fru_decode_product_info(product_area, &product))
+							fatal("Failed to decode product");
+						has_product = true;
+					}
+
+					free(buffer);
+				}
 				else {
 					fatal("The requested input file format is not supported");
```

Comment:
Now it looks more like `"Please specify the input file format"`

Reply:
You didn't actually resolve this, did you?

Reply:
Sorry, this edit got lost during the commit split